### PR TITLE
discordchatexporter-desktop: init at 2.44

### PR DESCRIPTION
**Pull request by Kek5chen**
## Things done

Added the missing GUI version of the discord chat exporter parallel to the discordchatexporter-cli that already exists. Source has been mostly copied and adjusted.

Also, **since discordchatexporter-cli is an orphan, I wouldn't mind adopting it as a maintainer.**

Implements #353780

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
@ofborg build discordchatexporter-desktop

File: pkgs/by-name/di/discordchatexporter-desktop/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  env.XDG_CONFIG_HOME = "$HOME/.config";
+
+  projectFile = "DiscordChatExporter.Gui/DiscordChatExporter.Gui.csproj";
+  nugetDeps = ./deps.nix;
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  # I hate this. Gladly suggest improvements... dotnet format doesn't want to do it
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/di/discordchatexporter-desktop/package.nix:29]: Please use a patch instead

File: pkgs/by-name/di/discordchatexporter-desktop/updater.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=./. -i bash -p curl jq common-updater-scripts
+set -eo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+new_version="$(curl -s "https://api.github.com/repos/tyrrrz/DiscordChatExporter/releases?per_page=1" | jq -r '.[0].name')"
```
Comment (SuperSandro2000) [inline pkgs/by-name/di/discordchatexporter-desktop/updater.sh:6]:
```suggestion
new_version="$(curl -s ${GITHUB_TOKEN:+-u ":$GITHUB_TOKEN"} "https://api.github.com/repos/tyrrrz/DiscordChatExporter/releases?per_page=1" | jq -r '.[0].name')"
```

File: pkgs/by-name/di/discordchatexporter-desktop/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  postFixup = ''
+    ln -s $out/bin/DiscordChatExporter $out/bin/discordchatexporter
+  '';
+
+  passthru = {
+    updateScript = ./updater.sh;
+    tests.version = testers.testVersion {
+      package = discordchatexporter-desktop;
+      version = "v${version}";
+    };
```
Comment (FliegendeWurst) [inline pkgs/by-name/di/discordchatexporter-desktop/package.nix:39]: This is a desktop application.
```suggestion
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**